packages/main/src/plugin/kubernetes/contexts-manager.ts

```diff
@@ -386,7 +386,13 @@ export class ContextsManager {
           sendGeneral: true,
           currentContext: this.kubeConfig.currentContext,
           resources: { pods: true },
-          update: state => state.resources.pods.push(obj),
+          update: state => {
```

Comment:
update is maybe juste updating the content of a pod ? like status or deleteTimestamps or idk which property. We should replace existing one with the new one provided.

Maybe state.resources.pods should be a Map<string, Pod> instead of an array, so we can quickly and efficiently update a pod by its id.

Reply:
Here, we are in the `onAdd` callback, when the informer sends an `ADD` event. In the case the resource is updated, this is handled below, in the `onUpdate` callback.

The map will be optimised for replacing objects, but as we want to send an array to the frontend, the extraction of the array from the map every time we want to send it won't be optimised anymore. I would prefer to keep this optimisation.